src/devices/src/virtio/rng/device.rs

```diff
@@ -110,19 +114,23 @@ impl Entropy {
         let mut used_any = false;
         while let Some(desc) = self.queues[RNG_QUEUE].pop(mem) {
             debug!("entropy: guest request for {} bytes of entropy", desc.len);
```

Comment:
I should've left this comment on an earlier commit, but I don't think this debug message is correct. `desc` is the descriptor chain head, so `len` is the length of the buffer to which the head descriptor points. However, if the chain has length longer than one, then the total number of random bytes requested is higher than `desc.len`, as its will be the sum of all `.len` of all descriptors in the chain, according to your `handle_one` impl.

Reply:
You are right. I have fixed this by introducing a helper type that parses the `DescriptorChain` into a something that derefs into slice of slice[u8]. We already use that (for other reasons) in the net device. I think it fixes the issue.